plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Library.java

```diff
@@ -26,4 +38,38 @@ public String getId() {
   public IPath getContainerPath() {
     return new Path(CONTAINER_PATH_PREFIX + "/" + id);
   }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public URI getSiteUri() {
+    return siteUri;
+  }
+
+  public void setSiteUri(URI siteUri) {
+    this.siteUri = siteUri;
+  }
+
+  public List<LibraryFile> getLibraryFiles() {
+    return libraryFiles;
+  }
+
+  public void setLibraryFiles(List<LibraryFile> libraryFiles) {
+    if (libraryFiles != null) {
+      this.libraryFiles = new LinkedList<>(libraryFiles);
```

Comment:
On modern hardware, array lists work better than linked lists for all operations, whatever the data structures textbooks say. See, for example,

https://kjellkod.wordpress.com/2012/02/25/why-you-should-never-ever-ever-use-linked-list-in-your-code-again/

TLDR: arrays make far better use of caches than linked data structures do. Furthermore, bulk memory moves on a modern CPU are usually faster than linear in the size of the data moved.